projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-crud.spec.ts

```diff
@@ -0,0 +1,1186 @@
+
+                UIInteractions.triggerKeyDownEvtUponElem('escape', cellDOM.nativeElement, true);
+                await wait(DEBOUNCETIME);
+                expect(cell.inEditMode).toBe(false, 'cannot exit edit mode after entering with enter');
+
+                UIInteractions.triggerKeyDownEvtUponElem('f2', cellDOM.nativeElement, true);
+                await wait(DEBOUNCETIME);
+                expect(cell.inEditMode).toBe(true, 'cannot enter edit mode with F2');
+
+                UIInteractions.triggerKeyDownEvtUponElem('escape', cellDOM.nativeElement, true);
+                await wait(DEBOUNCETIME);
+                expect(cell.inEditMode).toBe(false, 'cannot exit edit mode after entering with F2');
+            });
+
+            it('should be able to edit a tree-grid cell through UI', async() => {
+                const cell = treeGrid.getCellByColumn(0, 'ID');
+                const cellDomNumber = fix.debugElement.queryAll(By.css(CELL_CSS_CLASS))[0];
+
+                cellDomNumber.triggerEventHandler('dblclick', new Event('dblclick'));
+                await wait(DEBOUNCETIME);
+
+                expect(cell.inEditMode).toBe(true);
+                const editTemplate = cellDomNumber.query(By.css('input'));
+                expect(editTemplate).toBeDefined();
+
+                UIInteractions.sendInput(editTemplate, 146);
+                await wait(DEBOUNCETIME);
+                UIInteractions.triggerKeyDownEvtUponElem('enter', cellDomNumber.nativeElement, true);
+                await wait(DEBOUNCETIME);
+
+                expect(cell.inEditMode).toBe(false);
+                expect(parseInt(cell.value, 10)).toBe(146);
+                expect(editTemplate.nativeElement.type).toBe('number');
+                verifyCellValue(fix, 0, 'ID', '146');
+            });
+
+            it('should be able to edit a non-tree-grid cell through UI', async() => {
+                const cell = treeGrid.getCellByColumn(0, 'Name');
+                const cellDomNumber = fix.debugElement.queryAll(By.css(CELL_CSS_CLASS))[1];
+
+                cellDomNumber.triggerEventHandler('dblclick', new Event('dblclick'));
+                await wait(DEBOUNCETIME);
+
+                expect(cell.inEditMode).toBe(true);
+                const editTemplate = cellDomNumber.query(By.css('input'));
+                expect(editTemplate).toBeDefined();
+
+                UIInteractions.sendInput(editTemplate, 'Abc Def');
+                await wait(DEBOUNCETIME);
+                UIInteractions.triggerKeyDownEvtUponElem('enter', cellDomNumber.nativeElement, true);
+                await wait(DEBOUNCETIME);
+
+                expect(cell.inEditMode).toBe(false);
+                expect(cell.value).toBe('Abc Def');
+                expect(editTemplate.nativeElement.type).toBe('text');
+                verifyCellValue(fix, 0, 'Name', 'Abc Def');
+            });
+
+            it('should emit an event when editing a tree-grid cell through UI', async() => {
```

Comment:
Make sure to never use async(done) and call done() in an event handler. (Already resolved in this branch).

Reply:
ok